### PR TITLE
Set description when creating payment intents

### LINKED ISSUE #1807: Set `description` when creating payment intents
Right now, we aren't, so happiness engineers browsing the dashboard will see `ch_` charge IDs when they could be seeing something more like `Online payment for order 4567 for mystoredomain.com`

description

Edit: I've opened a related issue for iOS at https://github.com/woocommerce/woocommerce-ios/issues/4206

### PR DESCRIPTION
**Pull request by aprea**
Fixes #1807

#### Changes proposed in this Pull Request

Adds a description when creating payment intents. This description contains the order ID (if available), the store's domain name, and the blog ID (if available).

Follows a similar approach to https://github.com/woocommerce/woocommerce-ios/pull/4373

124856684-51a65d80-dfee-11eb-91aa-26114d0b8a01

#### Testing instructions

* Add items to the cart, complete checkout.
* Log into the Stripe dashboard and navigate to your connected account.
* Navigate to the list of payments.
* Observe a description is visible on the most recent transaction, e.g.
> Online Payment for Order #[Order ID] for [domain name] blog_id [blog ID]

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

**Comment by htdat** [comment]
👀

File: includes/wc-payment-api/class-wc-payments-api-client.php [hunk 1]
```diff
@@ -1431,4 +1435,27 @@ private function deserialize_intention_object_from_array( array $intention_array
...
 	}
+
+	/**
+	 * Returns a formatted intention description.
+	 *
+	 * @param  int $order_id The order ID.
+	 * @return string        A formatted intention description.
+	 */
+	private function get_intent_description( int $order_id ): string {
+		if ( empty( $order_id ) ) {
```
Comment (kalessil) [inline includes/wc-payment-api/class-wc-payments-api-client.php:1446]: We can express this more specific: `if ( 0 === $order_id )`. `empty` is not so great from maintaining perspective.

**Review by htdat — COMMENTED** [review]
👍 Tested and worked as the PR description. 

--- 

I am not sure if it's too late. However, I would change the description text if possible. 

Since we intend to serve Happiness Engineer here, I think we may use site domain and/or WPcom blog_id in the description. The main reason is that a site domain and a WP.com blog_id are much easier (and also unique) for HEs to locate the site than a site title. 

E.g: 

- `Online Payment for Order #69 for my-domain.com blog_id 123456`
- or `Online Payment for Order #69 for my-domain.com `

I may ask for feedback on this from @allendav (issue creator) and @DustinHartzler (HE ambassador mentioned in paJDYF-23i-p2)

**Comment by allendav** [comment]
I think including the blog ID is a wonderful idea

**Review by htdat — APPROVED** [review]
🚢 LGTM! 

✅ Tested with both localhost (dev) site and domain jurassic.tube. 
Markup on 2021-07-08 at 12:05:56

**Review by kalessil — APPROVED** [review]

**Pull request merged**